docs/testing/7_github_actions.md

```diff
@@ -0,0 +1,129 @@
+
+## Example for a Django project with Poetry
+
+The following code covers how to setup a workflow in Github Actions that will run your tests.
+It makes some assumptions, like the python version that it uses, poetry for package dependency and postgres for the database. However, you can infer what may be necessary according to your environment setup, as the workflow is neatly divided into steps.
+
+### Setup
+
+The basic setup is creating a .github folder in the root of your repository, then a workflows folder inside the .github folder, and then a django.yml file inside the workflows folder. Folder structure summary:
+
+- .github
+  - workflows
+    - django.yml
+
+### The Workflow
+
+The workflow is defined in the django.yml file.
+
+```yaml
+# The visible name that will appear in the github interface.
+name: Django Tests
+
+# You can specify branches and other actions, but the on: push directive is one of the most basics. Everytime there's a push on any branch, this workflow will be triggered and will execute its jobs.
+on: push
+
+# You can define environment variables on a global level for the workflow.
+# Here we define variables for the postgres database.
+# More information on how environment variables work here: https://docs.github.com/en/actions/learn-github-actions/variables#defining-environment-variables-for-a-single-workflow
+env:
+  POSTGRES_DB: postgres
+  POSTGRES_USER: postgres
+  POSTGRES_PASSWORD: postgres
+  POSTGRES_HOST: 127.0.0.1
+  POSTGRES_PORT: 15433
+
+jobs:
+  # Defines the name of the first and only job in this workflow.
+  django_tests:
+    # This job will run on a Ubuntu Linux runner with the Ubuntu version 22.04.
+    # Because this job requires a postgres database and it uses a docker container for that purpose. The Runner must be a Linux-based OS.
+    runs-on: ubuntu-22.04
+
+    # Services are Docker containers. For this job, we are using a postgres container for the database.
+    # See more information regarding the services here: https://docs.github.com/en/actions/using-containerized-services/about-service-containers
+    services:
+      # A name for the service.
+      postgres:
+        # Use postgres version 15.
+        image: "postgres:15"
+        # Define the variables for the postgres container.
+        env:
+          POSTGRES_PASSWORD: ${{env.POSTGRES_PASSWORD}}
+          POSTGRES_USER: ${{env.POSTGRES_USER}}
+          POSTGRES_DB: ${{env.POSTGRES_DB}}
+        # Note that we didn't use ${{env.POSTGRES_PORT}} in the ports section. That's because you don't have access to the env object in the ports section.
+        # Read more about this here: https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability
+        # TODO: If anyone knows if there's a way to use the env value in the ports, feel free to make a PR.
```

Comment:
I want to look into this, how can I test any changes I make to the github actions? does it automatically get updated on my branch?

Reply:
If you make a change to a github action, you can only see the results of your change when you trigger that action. The trigger in this example is a push.

I didn't add the github action as an actual github action, it's just a documentation, but maybe I should add it to this repo for demonstration purposes.

There are tools like [act](https://github.com/nektos/act) that allow you to test a github action locally (read: without having to trigger anything in the repository), but they don't work perfectly. However, act doesn't work with any service; in this example, there's a postgres service and that doesn't work with act.